File: solutions/c/isogram/1/isogram.c
#include "isogram.h"
#include <stdbool.h>
#include <ctype.h>
#include <stddef.h>

bool is_isogram(const char phrase[]) {
    if (phrase == NULL) return false;

    bool seen[26] = { false };

    for (const char *p = phrase; *p; ++p) {
        if (*p == ' ' || *p == '-') {
            continue;
        }

        if (isalpha(*p)) {
            int index = tolower(*p) - 'a';
            if (seen[index]) {
                return false;
            }
            seen[index] = true;
        }
    }

    return true;
}

File: solutions/go/isogram/1/isogram.go
package isogram

import "strings"

func IsIsogram(word string) bool {
    word = strings.ToLower(word)
	for i := 0; i < len(word) - 1; i++ {
        for j := i + 1; j < len(word); j++ {
            if word[i] == word[j] && int(word[i]) >= int('a') && int(word[i]) <= int('z') {
                return false
            }
        }
    }
    return true
}
